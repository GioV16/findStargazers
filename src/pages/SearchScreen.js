import React, {useState} from 'react';
import {StyleSheet, View, Text, Image, ScrollView, Alert, StatusBar} from 'react-native';
import {getStar} from '../api';
import {Button} from '../components/Button';
import {ParamInput} from '../components/ParamInput';
import {COLOR} from '../costants/colors';
import {FONT} from '../costants/fonts';
import {TEXTS} from '../costants/texts';

/*
  @props : 
    handleStarsFounded = we use it for pass and handle the stargazlers in father
*/
export const SearchScreen = props => {
  // const [owner, setOwner] = useState('octocat');
  // const [repo, setRepo] = useState('Spoon-Knife');

  const [owner, setOwner] = useState('');
  const [repo, setRepo] = useState('');

  /* Callback to our api, it will pass a list of stargazers if 
  the params match the request, or will handle the error throw
  with an alert */
  const findStargazers = async () => {
    try {
      const stars = await getStar(owner, repo);
      if (stars) {
        props?.handleStarsFounded(stars);
      }
    } catch (error) {
      Alert.alert(
        TEXTS.SEARCH_SCREEN.ERROR_TITLE,
        TEXTS.SEARCH_SCREEN.ERROR_MESSAGE,
      );
      setOwner('');
      setRepo('');
    }
  };

  return (
    <View style={styles.mainContainer}>
      <ScrollView contentContainerStyle={styles.bodyContainer}>
        <View style={styles.presentationSection}>
          <View style={styles.presentationSectionImage}>
            <Image
              width={50}
              height={50}
              source={require('./../assets/icons/stargazing.png')}></Image>
          </View>
          <Text style={styles.presentationSectionText}>
            {TEXTS.SEARCH_SCREEN.PRESENTATION}
          </Text>
        </View>
        <View>
          <ParamInput
            placeholder={TEXTS.SEARCH_SCREEN.PLACEHOLDER_OWNER}
            value={owner}
            onChangeText={setOwner}
          />
          <ParamInput
            placeholder={TEXTS.SEARCH_SCREEN.PLACEHOLDER_REPO}
            value={repo}
            onChangeText={setRepo}
          />
        </View>
        <View style={styles.footerContainer}>
          <Button
            onPressButton={findStargazers}
            buttonText={TEXTS.BUTTON.SEARCH}
          />
        </View>
      </ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  mainContainer: {
    flex: 1,
    backgroundColor: COLOR.BACKGROUND_COLOR,
  },
  bodyContainer: {
    flex: 1,
    justifyContent: 'center',
  },
  presentationSection: {
    paddingVertical: 20,
    width: '80%',
    alignItems: 'center',
    alignSelf: 'center',
  },
  presentationSectionText: {
    ...FONT.PRESENTATION,
  },
  presentationSectionImage: {
    paddingBottom: 20,
  },
  footerContainer: {
    alignItems: 'center',
    justifyContent: 'flex-end',
    paddingVertical: 50,
  },
});
