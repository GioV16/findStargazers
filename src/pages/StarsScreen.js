import React from 'react';
import {StyleSheet, View, ScrollView} from 'react-native';
import {Button} from '../components/Button';
import {StarsListItem} from '../components/StarsListItem';
import {COLOR} from '../costants/colors';
import {TEXTS} from '../costants/texts';

/*
  @props : 
    starList = starList passed from caller;
    closeStarsScreen = CTA on close button screen
*/
export const StarsScreen = props => {
  return (
    <View style={styles.mainContainer}>
      <ScrollView contentContainerStyle={styles.bodyContainer}>
        {props.starList.map(star => {
          return <StarsListItem key={star.username} starItem={star} />;
        })}
      </ScrollView>
      <View style={styles.footerContainer}>
        <Button
          onPressButton={props.closeStarsScreen}
          buttonText={TEXTS.BUTTON.RESEARCH}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  mainContainer: {
    flex: 1,
    backgroundColor: COLOR.BACKGROUND_COLOR,
  },
  bodyContainer: {
    justifyContent: 'center',
  },
  footerContainer: {
    alignItems: 'center',
    justifyContent: 'flex-end',
    paddingVertical: 20,
  },
});
