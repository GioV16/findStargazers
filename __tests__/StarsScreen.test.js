import 'react-native';
import React from 'react';
import { StarsScreen } from '../src/pages/StarsScreen';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<StarsScreen 
    starList={[]}
  />);
  expect(tree).toMatchSnapshot();
});
