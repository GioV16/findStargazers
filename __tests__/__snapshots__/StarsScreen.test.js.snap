// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#CDC6A5",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "justifyContent": "center",
      }
    }
  >
    <View />
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "flex-end",
        "paddingVertical": 20,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#313E50",
          "borderRadius": 5,
          "justifyContent": "center",
          "padding": 10,
          "width": "80%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "flexGrow": 1,
            "fontSize": 17,
            "letterSpacing": 0.2,
            "textAlign": "center",
          }
        }
      >
        Back to the airship
      </Text>
    </View>
  </View>
</View>
`;
